src/components/form/index.tsx

```diff
@@ -26,17 +28,28 @@ export class SmoothlyForm implements Changeable, Clearable, Submittable {
 	@Event() notice: EventEmitter<Notice>
 
 	@Prop({ mutable: true, reflect: true }) changed = false
-	private listeners: { changed?: ((parent: Changeable) => Promise<void>)[] } = {}
+	private listeners: {
+		changed?: ((parent: Changeable) => Promise<void>)[]
+		readonly?: ((parent: Editable) => Promise<void>)[]
+	} = {}
 
 	listen(property: "changed", listener: (parent: Changeable) => Promise<void>): void {
```

Comment:
this listener will get a better name when i implement the clear button listening to "readonly"-status of form and input, not included here because that would mean updating Changeable-type and then we have a snowball